src/adagrams.js

```diff
@@ -1,15 +1,121 @@
+const LETTER_POOL = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
 export const drawLetters = () => {
-  // Implement this method for wave 1
+
+  let drawedLetters = [];
+  const numAllowedinHand = 10;
+  let allLetters = [];
+
+  for (const [key, value] of Object.entries(LETTER_POOL)){
+    for (let i = 0; i < value; i++){
+      allLetters.push(key);
+    };
+  };
+
+  while (drawedLetters.length < numAllowedinHand){
+    let randomNumber = Math.floor(Math.random() * allLetters.length);
+    drawedLetters.push(allLetters[randomNumber]);
+    allLetters.splice(randomNumber,1);
+  
+  };
+  
+  return drawedLetters;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
   // Implement this method for wave 2
+  let lettersFreq = {};
+
+  for (const letter of input){
+      lettersFreq[letter] = (lettersFreq[letter] || 0) + 1;
+  };
+
+  let letterHandFreq = {};
+  for (const letter of lettersInHand){
+    letterHandFreq[letter] = (letterHandFreq[letter] || 0) + 1;
+  };
```

Comment:
What if you created a helper function that created an object and you could use the helper function in multiple places to reduce repetition? Maybe a helper called `createFrequencyObject`

```suggestion
  let lettersFreq = createFrequencyObject(input);
  let letterHandFreq = createFrequencyObject(lettersInHand);
```

Using a helper function for this logic would make `usesAvailableLetters` more single responsibility too.